### DISCUSSION TITLE
concat with 2d arrays with same dim along x and y

### DISCUSSION THREAD

**Discussion by jvail** [pydata/xarray#5213] [Q&A] [answered]
I have datasets with variables that have the same dimension along axis 0 and 1. Something like this:

```
xr.DataArray(
    np.arange(9).reshape(3, 3), [('x', [0, 1, 2]), ('x', [0, 1, 2])]
)
```

I am trying to concat multiple ds but it fails with

`ValueError: axes don't match array`

Example:
```
xr.concat([
    xr.DataArray(np.arange(9).reshape(3, 3), [('x', [0, 1, 2]), ('x', [0, 1, 2])]), 
    xr.DataArray(np.arange(4).reshape(2, 2), [('x', [1, 2]), ('x', [1, 2])])
], dim='new_dim')
```

But if I rename the second x to x1 it works. Might this be a bug or is it intended?

**Comment by keewis** [comment, marked as answer]
that's expected, arrays with repeated dimensions are difficult to support (see #1378) and essentially broken right now. We could improve the error message, though, either by rejecting repeated dimension names in the constructors or where this breaks something (option A would be easier but a breaking change).

**Reply by jvail** [reply]
Thank you, wasn't aware of the existing issue.
